rocketpy/mathutils/function.py

```diff
@@ -165,20 +174,20 @@ def set_source(self, source):
             source = source.get_source()
         # Import CSV if source is a string or Path and convert values to ndarray
         if isinstance(source, (str, Path)):
-            # Read file and check for headers
-            with open(source, mode="r") as f:
-                first_line = f.readline()
-            # If headers are found...
-            if first_line[0] in ['"', "'"]:
-                # Headers available
-                first_line = first_line.replace('"', " ").replace("'", " ")
-                first_line = first_line.split(" , ")
-                self.set_inputs(first_line[0])
-                self.set_outputs(first_line[1:])
-                source = np.loadtxt(source, delimiter=",", skiprows=1, dtype=float)
-            # if headers are not found
-            else:
-                source = np.loadtxt(source, delimiter=",", dtype=float)
+            with open(source, "r") as file:
+                first_line = file.readline()
+                try:
+                    # Try to convert the first line to floats
+                    np.array([float(item) for item in first_line.split(",")])
+                    # If successful, no headers are present
+                    file.seek(0)  # Reset file read position
+                    source = np.loadtxt(file, delimiter=",", dtype=float)
```

Comment:
I might have misunderstood something, but these checks before `np.loadtxt` seem pointless:

- Why to cast every item in the first line to a `float`, return to the beginning, and then apply `np.loadtxt` that basically will try to cast all the items to a `float`?
- Isn't it the same as using `np.loadtxt` in the first place and, if there is an error, that means there is a header.

Furthermore, I do think that should be another exception layer in case the `np.gentxt` fails. That means the input is malformed and the user should be informed.

Of course, I am not trying to bias you to the same logic I did in this PR at `__check_user_inputs`, but it seems that there may be some unnecessary operations here.

Reply:
I think your approach is better.